File: 2020/d/record-breaker/v1.kt
import kotlin.math.max

fun main() {
    val T = readLine()!!.toInt()
    repeat(T) { t ->
        val N = readLine()!!.toInt()
        val V = readLine()!!.split(' ').map(String::toInt)
        val metNums = mutableSetOf<Int>()
        var currentMax = V[0]
        var count = 0
        for (i in 0 until N) {
            currentMax = max(currentMax, V[i])
            if (V[i] != currentMax || V[i] in metNums) {
                continue
            }
            metNums.add(V[i])
            if (i == N - 1 || V[i] > V[i + 1]) {
                count++
            }
        }
        println("Case #${t + 1}: $count")

    }
}

File: 2020/d/record-breaker/v1.py

def printRes(order, result):
    print(f"Case #{order + 1}: {result}")

def readln():
    return input().strip()
    
def readInt():
    return int(input().strip())

def readInts():
    return list(map(int, input().split()))

def solve(order):
    N = readInt()
    V = readInts()
    met_nums = set()
    current_max = V[0]
    count = 0
    for i in range(N):
        current_max = max(current_max, V[i])
        if V[i] != current_max or V[i] in met_nums:
            continue
        met_nums.add(V[i])
        if i == N - 1 or V[i] > V[i + 1]:
            count += 1

    printRes(order, count)

if __name__ == "__main__":

    T = readInt()
    for t in range(T):
        solve(t)
